internal/discovery/pods_lister.go

```diff
@@ -0,0 +1,34 @@
+package discovery
+
+import (
+	"time"
+
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/listers/core/v1"
+)
+
+type PodsListerConfig struct {
+	// Namespace can be used to restric the search to a particular namespace.
+	Namespace string
+	// Client is the Kubernetes client.Interface used to build informers.
+	Client kubernetes.Interface
+}
+
+func NewPodsLister(config PodsListerConfig) (v1.PodLister, chan<- struct{}) {
+	// Arbitrary value, same used in Prometheus.
+	resyncDuration := 10 * time.Minute
+	factory := informers.NewSharedInformerFactoryWithOptions(
+		config.Client,
+		resyncDuration,
+		informers.WithNamespace(config.Namespace),
```

Comment:
if we are specifying the namespace here, should we return a "v1.PodNamespaceLister"?

Reply:
yes make sense i did not looked at that